tests/platform/mellanox/check_sysfs.py

```diff
@@ -34,46 +34,138 @@ def check_sysfs(dut):
     dut_hwsku = dut.facts["hwsku"]
     from common.mellanox_data import SWITCH_MODELS
     fan_count = SWITCH_MODELS[dut_hwsku]["fans"]["number"]
-
-    if SWITCH_MODELS[dut_hwsku]["fans"]["hot_swappable"]:
-        fan_status_list = ["/var/run/hw-management/thermal/fan%d_status" % fan_id for fan_id in range(1, fan_count + 1)]
-        for fan_status in fan_status_list:
+    
+    for fan_id in range(1, fan_count + 1):
+        if SWITCH_MODELS[dut_hwsku]["fans"]["hot_swappable"]:
+            fan_status = "/var/run/hw-management/thermal/fan{}_status".format(fan_id)
             fan_status_content = dut.command("cat %s" % fan_status)
             assert fan_status_content["stdout"] == "1", "Content of %s is not 1" % fan_status
-
-    fan_fault_list = ["/var/run/hw-management/thermal/fan%d_fault" % fan_id for fan_id in range(1, fan_count + 1)]
-    for fan_fault in fan_fault_list:
+        
+        fan_fault = "/var/run/hw-management/thermal/fan{}_fault".format(fan_id)
         fan_fault_content = dut.command("cat %s" % fan_fault)
         assert fan_fault_content["stdout"] == "0", "Content of %s is not 0" % fan_fault
-
-    fan_min_list = ["/var/run/hw-management/thermal/fan%d_min" % fan_id for fan_id in range(1, fan_count + 1)]
-    for fan_min in fan_min_list:
+        
+        fan_min = "/var/run/hw-management/thermal/fan{}_min".format(fan_id)
         try:
             fan_min_content = dut.command("cat %s" % fan_min)
             fan_min_speed = int(fan_min_content["stdout"])
             assert fan_min_speed > 0, "Bad fan minimum speed: %s" % str(fan_min_speed)
         except Exception as e:
-            assert False, "Get content from %s failed, exception: %s" % (fan_min, repr(e))
+            assert "Get content from %s failed, exception: %s" % (fan_min, repr(e))
 
-    fan_max_list = ["/var/run/hw-management/thermal/fan%d_max" % fan_id for fan_id in range(1, fan_count + 1)]
-    for fan_max in fan_max_list:
+        fan_max = "/var/run/hw-management/thermal/fan{}_max".format(fan_id)
         try:
             fan_max_content = dut.command("cat %s" % fan_max)
             fan_max_speed = int(fan_max_content["stdout"])
             assert fan_max_speed > 10000, "Bad fan maximum speed: %s" % str(fan_max_speed)
         except Exception as e:
-            assert False, "Get content from %s failed, exception: %s" % (fan_max, repr(e))
+            assert "Get content from %s failed, exception: %s" % (fan_max, repr(e))
+
+        fan_speed_set = "/var/run/hw-management/thermal/fan{}_speed_set".format(fan_id)
+        fan_speed_set_content = dut.command("cat %s" % fan_speed_set)
+        assert fan_speed_set_content["stdout"] == "153", "Fan speed should be set to 60%, 153/255"
```

Comment:
how is 153 calculated? same for all platforms or should be calculated differently?

Reply:
153 is 60% of 255,  when the hw-mgmt thermal control suspended the fan speed will be set running at 60% speed, this is common for all Mellanox platforms.

Reply:
OK